index.js

```diff
@@ -79,17 +80,20 @@ class Listr {
 
 		this._checkAll(context);
 
-		let tasks;
-		if (this._options.concurrent === true) {
-			tasks = Promise.all(this._tasks.map(task => runTask(task, context, errors)));
+		let concurrency;
+		if (!this._options.concurrent) {
```

Comment:
Also wondering, wouldn't it be cleaner if we did all of this in the `constructor` instead? And thus set `this._options.concurrent` to the correct value for `p-map`.